### PR TITLE
Memleaks (fix #11389)

### LINKED ISSUE #11389: memory leaks
```
=================================================================
==728==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 3238 byte(s) in 2030 object(s) allocated from:
    #0 0x7ffb41801f01 in __interceptor_strdup /build/gcc/src/gcc/libsanitizer/asan/asan_interceptors.cc:405
    #1 0x7ffb38ecc5db in r_strbuf_drain /usr/src/debug/radare2-git/libr/util/strbuf.c:172
    #2 0x7ffb40acf6ec in r_core_anal_hasrefs_to_depth /usr/src/debug/radare2-git/libr/core/core.c:1920
    #3 0x7ffb40ad09bb in r_core_anal_hasrefs /usr/src/debug/radare2-git/libr/core/core.c:1788
    #4 0x7ffb38eb5718 in r_print_hexdump /usr/src/debug/radare2-git/libr/util/print.c:1046
    #5 0x7ffb40c7e8f6 in cmd_print /usr/src/debug/radare2-git/libr/core/cmd_print.c:5242
    #6 0x7ffb40d981dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #7 0x7ffb40cb25b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #8 0x7ffb40bc8929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #9 0x7ffb40bc9af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #10 0x7ffb40be9e5e in r_core_cmd0 /usr/src/debug/radare2-git/libr/core/cmd.c:3769
    #11 0x7ffb40cff403 in visual_refresh /usr/src/debug/radare2-git/libr/core/visual.c:3156
    #12 0x7ffb40e59fb0 in r_core_task_enqueue_oneshot /usr/src/debug/radare2-git/libr/core/task.c:446
    #13 0x7ffb40cedc93 in visual_refresh_oneshot /usr/src/debug/radare2-git/libr/core/visual.c:3184
    #14 0x7ffb4005923f in resizeWin /usr/src/debug/radare2-git/libr/cons/cons.c:346
    #15 0x7ffb4007c68c in r_cons_readchar /usr/src/debug/radare2-git/libr/cons/input.c:507
    #16 0x7ffb40d0e5ef in r_core_visual /usr/src/debug/radare2-git/libr/core/visual.c:3342
    #17 0x7ffb40b183b2 in cmd_visual /usr/src/debug/radare2-git/libr/core/cmd.c:1292
    #18 0x7ffb40d981dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #19 0x7ffb40cb25b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #20 0x7ffb40bc8929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #21 0x7ffb40bc9af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #22 0x7ffb40ad8fa6 in r_core_prompt_exec /usr/src/debug/radare2-git/libr/core/core.c:2568
    #23 0x55b27bab57ab in main /usr/src/debug/radare2-git/binr/radare2/radare2.c:1406
    #24 0x7ffb37e13222 in __libc_start_main (/usr/lib/libc.so.6+0x24222)

Direct leak of 3238 byte(s) in 2030 object(s) allocated from:
    #0 0x7ffb41801f01 in __interceptor_strdup /build/gcc/src/gcc/libsanitizer/asan/asan_interceptors.cc:405
    #1 0x7ffb38ecc5db in r_strbuf_drain /usr/src/debug/radare2-git/libr/util/strbuf.c:172
    #2 0x7ffb40acf6ec in r_core_anal_hasrefs_to_depth /usr/src/debug/radare2-git/libr/core/core.c:1920
    #3 0x7ffb40ad09bb in r_core_anal_hasrefs /usr/src/debug/radare2-git/libr/core/core.c:1788
    #4 0x7ffb38eb5718 in r_print_hexdump /usr/src/debug/radare2-git/libr/util/print.c:1046
    #5 0x7ffb40c7e8f6 in cmd_print /usr/src/debug/radare2-git/libr/core/cmd_print.c:5242
    #6 0x7ffb40d981dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #7 0x7ffb40cb25b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #8 0x7ffb40bc8929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #9 0x7ffb40bc9af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #10 0x7ffb40be9e5e in r_core_cmd0 /usr/src/debug/radare2-git/libr/core/cmd.c:3769
    #11 0x7ffb40cff403 in visual_refresh /usr/src/debug/radare2-git/libr/core/visual.c:3156
    #12 0x7ffb40d0ebe9 in r_core_visual /usr/src/debug/radare2-git/libr/core/visual.c:3330
    #13 0x7ffb40b183b2 in cmd_visual /usr/src/debug/radare2-git/libr/core/cmd.c:1292
    #14 0x7ffb40d981dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #15 0x7ffb40cb25b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #16 0x7ffb40bc8929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #17 0x7ffb40bc9af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #18 0x7ffb40ad8fa6 in r_core_prompt_exec /usr/src/debug/radare2-git/libr/core/core.c:2568
    #19 0x55b27bab57ab in main /usr/src/debug/radare2-git/binr/radare2/radare2.c:1406
    #20 0x7ffb37e13222 in __libc_start_main (/usr/lib/libc.so.6+0x24222)

Direct leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7ffb418b9231 in __interceptor_calloc /build/gcc/src/gcc/libsanitizer/asan/asan_malloc_linux.cc:95
    #1 0x7ffb38e50191 in r_list_new /usr/src/debug/radare2-git/libr/util/list.c:155
    #2 0x7ffb3d8d9b24 in r_anal_fcn_list_new /usr/src/debug/radare2-git/libr/anal/fcn.c:305
    #3 0x7ffb40db813c in core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:723
    #4 0x7ffb40db3eb5 in r_core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:1551
    #5 0x7ffb40db5ce5 in r_anal_analyze_fcn_refs /usr/src/debug/radare2-git/libr/core/canal.c:534
    #6 0x7ffb40db7f75 in core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:760
    #7 0x7ffb40db3eb5 in r_core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:1551
    #8 0x7ffb40c5cc04 in cmd_anal_fcn /usr/src/debug/radare2-git/libr/core/cmd_anal.c:2842
    #9 0x7ffb40c63c76 in cmd_anal /usr/src/debug/radare2-git/libr/core/cmd_anal.c:7366
    #10 0x7ffb40d981dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #11 0x7ffb40cb5225 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2857
    #12 0x7ffb40bc8929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #13 0x7ffb40bc9af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #14 0x7ffb40bcef5e in r_core_cmdf /usr/src/debug/radare2-git/libr/core/cmd.c:3763
    #15 0x7ffb40dfa8d0 in r_core_visual_define /usr/src/debug/radare2-git/libr/core/vmenus.c:3194
    #16 0x7ffb40d0498e in r_core_visual_cmd /usr/src/debug/radare2-git/libr/core/visual.c:2069
    #17 0x7ffb40d0ec7e in r_core_visual /usr/src/debug/radare2-git/libr/core/visual.c:3354
    #18 0x7ffb40b183b2 in cmd_visual /usr/src/debug/radare2-git/libr/core/cmd.c:1292
    #19 0x7ffb40d981dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #20 0x7ffb40cb25b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #21 0x7ffb40bc8929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #22 0x7ffb40bc9af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #23 0x7ffb40ad8fa6 in r_core_prompt_exec /usr/src/debug/radare2-git/libr/core/core.c:2568
    #24 0x55b27bab57ab in main /usr/src/debug/radare2-git/binr/radare2/radare2.c:1406
    #25 0x7ffb37e13222 in __libc_start_main (/usr/lib/libc.so.6+0x24222)

Indirect leak of 24 byte(s) in 1 object(s) allocated from:
    #0 0x7ffb418b9019 in __interceptor_malloc /build/gcc/src/gcc/libsanitizer/asan/asan_malloc_linux.cc:86
    #1 0x7ffb38e502a3 in r_list_append /usr/src/debug/radare2-git/libr/util/list.c:183
    #2 0x7ffb40db80ac in core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:725
    #3 0x7ffb40db3eb5 in r_core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:1551
    #4 0x7ffb40db5ce5 in r_anal_analyze_fcn_refs /usr/src/debug/radare2-git/libr/core/canal.c:534
    #5 0x7ffb40db7f75 in core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:760
    #6 0x7ffb40db3eb5 in r_core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:1551
    #7 0x7ffb40c5cc04 in cmd_anal_fcn /usr/src/debug/radare2-git/libr/core/cmd_anal.c:2842
    #8 0x7ffb40c63c76 in cmd_anal /usr/src/debug/radare2-git/libr/core/cmd_anal.c:7366
    #9 0x7ffb40d981dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #10 0x7ffb40cb5225 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2857
    #11 0x7ffb40bc8929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #12 0x7ffb40bc9af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #13 0x7ffb40bcef5e in r_core_cmdf /usr/src/debug/radare2-git/libr/core/cmd.c:3763
    #14 0x7ffb40dfa8d0 in r_core_visual_define /usr/src/debug/radare2-git/libr/core/vmenus.c:3194
    #15 0x7ffb40d0498e in r_core_visual_cmd /usr/src/debug/radare2-git/libr/core/visual.c:2069
    #16 0x7ffb40d0ec7e in r_core_visual /usr/src/debug/radare2-git/libr/core/visual.c:3354
    #17 0x7ffb40b183b2 in cmd_visual /usr/src/debug/radare2-git/libr/core/cmd.c:1292
    #18 0x7ffb40d981dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #19 0x7ffb40cb25b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #20 0x7ffb40bc8929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #21 0x7ffb40bc9af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #22 0x7ffb40ad8fa6 in r_core_prompt_exec /usr/src/debug/radare2-git/libr/core/core.c:2568
    #23 0x55b27bab57ab in main /usr/src/debug/radare2-git/binr/radare2/radare2.c:1406
    #24 0x7ffb37e13222 in __libc_start_main (/usr/lib/libc.so.6+0x24222)

SUMMARY: AddressSanitizer: 6532 byte(s) leaked in 4062 allocation(s).

=================================================================
==746==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 2239 byte(s) in 1624 object(s) allocated from:
    #0 0x7ff88a08af01 in __interceptor_strdup /build/gcc/src/gcc/libsanitizer/asan/asan_interceptors.cc:405
    #1 0x7ff8817555db in r_strbuf_drain /usr/src/debug/radare2-git/libr/util/strbuf.c:172
    #2 0x7ff8893586ec in r_core_anal_hasrefs_to_depth /usr/src/debug/radare2-git/libr/core/core.c:1920
    #3 0x7ff8893599bb in r_core_anal_hasrefs /usr/src/debug/radare2-git/libr/core/core.c:1788
    #4 0x7ff88173e718 in r_print_hexdump /usr/src/debug/radare2-git/libr/util/print.c:1046
    #5 0x7ff8895078f6 in cmd_print /usr/src/debug/radare2-git/libr/core/cmd_print.c:5242
    #6 0x7ff8896211dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #7 0x7ff88953b5b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #8 0x7ff889451929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #9 0x7ff889452af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #10 0x7ff889472e5e in r_core_cmd0 /usr/src/debug/radare2-git/libr/core/cmd.c:3769
    #11 0x7ff889588403 in visual_refresh /usr/src/debug/radare2-git/libr/core/visual.c:3156
    #12 0x7ff8896e2fb0 in r_core_task_enqueue_oneshot /usr/src/debug/radare2-git/libr/core/task.c:446
    #13 0x7ff889576c93 in visual_refresh_oneshot /usr/src/debug/radare2-git/libr/core/visual.c:3184
    #14 0x7ff8888e223f in resizeWin /usr/src/debug/radare2-git/libr/cons/cons.c:346
    #15 0x7ff88890568c in r_cons_readchar /usr/src/debug/radare2-git/libr/cons/input.c:507
    #16 0x7ff8895975ef in r_core_visual /usr/src/debug/radare2-git/libr/core/visual.c:3342
    #17 0x7ff8893a13b2 in cmd_visual /usr/src/debug/radare2-git/libr/core/cmd.c:1292
    #18 0x7ff8896211dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #19 0x7ff88953b5b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #20 0x7ff889451929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #21 0x7ff889452af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #22 0x7ff889361fa6 in r_core_prompt_exec /usr/src/debug/radare2-git/libr/core/core.c:2568
    #23 0x557d4a62d7ab in main /usr/src/debug/radare2-git/binr/radare2/radare2.c:1406
    #24 0x7ff88069c222 in __libc_start_main (/usr/lib/libc.so.6+0x24222)

Direct leak of 2239 byte(s) in 1624 object(s) allocated from:
    #0 0x7ff88a08af01 in __interceptor_strdup /build/gcc/src/gcc/libsanitizer/asan/asan_interceptors.cc:405
    #1 0x7ff8817555db in r_strbuf_drain /usr/src/debug/radare2-git/libr/util/strbuf.c:172
    #2 0x7ff8893586ec in r_core_anal_hasrefs_to_depth /usr/src/debug/radare2-git/libr/core/core.c:1920
    #3 0x7ff8893599bb in r_core_anal_hasrefs /usr/src/debug/radare2-git/libr/core/core.c:1788
    #4 0x7ff88173e718 in r_print_hexdump /usr/src/debug/radare2-git/libr/util/print.c:1046
    #5 0x7ff8895078f6 in cmd_print /usr/src/debug/radare2-git/libr/core/cmd_print.c:5242
    #6 0x7ff8896211dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #7 0x7ff88953b5b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #8 0x7ff889451929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #9 0x7ff889452af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #10 0x7ff889472e5e in r_core_cmd0 /usr/src/debug/radare2-git/libr/core/cmd.c:3769
    #11 0x7ff889588403 in visual_refresh /usr/src/debug/radare2-git/libr/core/visual.c:3156
    #12 0x7ff889597be9 in r_core_visual /usr/src/debug/radare2-git/libr/core/visual.c:3330
    #13 0x7ff8893a13b2 in cmd_visual /usr/src/debug/radare2-git/libr/core/cmd.c:1292
    #14 0x7ff8896211dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #15 0x7ff88953b5b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #16 0x7ff889451929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #17 0x7ff889452af1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #18 0x7ff889361fa6 in r_core_prompt_exec /usr/src/debug/radare2-git/libr/core/core.c:2568
    #19 0x557d4a62d7ab in main /usr/src/debug/radare2-git/binr/radare2/radare2.c:1406
    #20 0x7ff88069c222 in __libc_start_main (/usr/lib/libc.so.6+0x24222)

SUMMARY: AddressSanitizer: 4478 byte(s) leaked in 3248 allocation(s).

=================================================================
==882==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x7f3f73ada231 in __interceptor_calloc /build/gcc/src/gcc/libsanitizer/asan/asan_malloc_linux.cc:95
    #1 0x7f3f6b071191 in r_list_new /usr/src/debug/radare2-git/libr/util/list.c:155
    #2 0x7f3f6fafab24 in r_anal_fcn_list_new /usr/src/debug/radare2-git/libr/anal/fcn.c:305
    #3 0x7f3f72fd913c in core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:723
    #4 0x7f3f72fd4eb5 in r_core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:1551
    #5 0x7f3f72fd6ce5 in r_anal_analyze_fcn_refs /usr/src/debug/radare2-git/libr/core/canal.c:534
    #6 0x7f3f72fd8f75 in core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:760
    #7 0x7f3f72fd4eb5 in r_core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:1551
    #8 0x7f3f72e7dc04 in cmd_anal_fcn /usr/src/debug/radare2-git/libr/core/cmd_anal.c:2842
    #9 0x7f3f72e84c76 in cmd_anal /usr/src/debug/radare2-git/libr/core/cmd_anal.c:7366
    #10 0x7f3f72fb91dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #11 0x7f3f72ed35b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #12 0x7f3f72de9929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #13 0x7f3f72deaaf1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #14 0x7f3f72cf9fa6 in r_core_prompt_exec /usr/src/debug/radare2-git/libr/core/core.c:2568
    #15 0x55f2042e47ab in main /usr/src/debug/radare2-git/binr/radare2/radare2.c:1406
    #16 0x7f3f6a034222 in __libc_start_main (/usr/lib/libc.so.6+0x24222)

Indirect leak of 24 byte(s) in 1 object(s) allocated from:
    #0 0x7f3f73ada019 in __interceptor_malloc /build/gcc/src/gcc/libsanitizer/asan/asan_malloc_linux.cc:86
    #1 0x7f3f6b0712a3 in r_list_append /usr/src/debug/radare2-git/libr/util/list.c:183
    #2 0x7f3f72fd90ac in core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:725
    #3 0x7f3f72fd4eb5 in r_core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:1551
    #4 0x7f3f72fd6ce5 in r_anal_analyze_fcn_refs /usr/src/debug/radare2-git/libr/core/canal.c:534
    #5 0x7f3f72fd8f75 in core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:760
    #6 0x7f3f72fd4eb5 in r_core_anal_fcn /usr/src/debug/radare2-git/libr/core/canal.c:1551
    #7 0x7f3f72e7dc04 in cmd_anal_fcn /usr/src/debug/radare2-git/libr/core/cmd_anal.c:2842
    #8 0x7f3f72e84c76 in cmd_anal /usr/src/debug/radare2-git/libr/core/cmd_anal.c:7366
    #9 0x7f3f72fb91dc in r_cmd_call /usr/src/debug/radare2-git/libr/core/cmd_api.c:237
    #10 0x7f3f72ed35b2 in r_core_cmd_subst_i /usr/src/debug/radare2-git/libr/core/cmd.c:2901
    #11 0x7f3f72de9929 in r_core_cmd_subst /usr/src/debug/radare2-git/libr/core/cmd.c:1930
    #12 0x7f3f72deaaf1 in r_core_cmd /usr/src/debug/radare2-git/libr/core/cmd.c:3605
    #13 0x7f3f72cf9fa6 in r_core_prompt_exec /usr/src/debug/radare2-git/libr/core/core.c:2568
    #14 0x55f2042e47ab in main /usr/src/debug/radare2-git/binr/radare2/radare2.c:1406
    #15 0x7f3f6a034222 in __libc_start_main (/usr/lib/libc.so.6+0x24222)

SUMMARY: AddressSanitizer: 56 byte(s) leaked in 2 allocation(s).
```

### PR DESCRIPTION
**Pull request by ret2libc**

### CODE REVIEW

**Comment by radare** [comment]
21 XX and one bold word `libr/egg/egg.c:                         eprintf ("Fuck this shit. Cannot patch outside\n");` pls fix that error message in this pr too

**Comment by ret2libc** [comment]
R2r https://github.com/radare/radare2-regressions/pull/1505

File: libr/anal/fcn.c [hunk 1]
```diff
@@ -302,12 +302,7 @@ R_API RAnalFunction *r_anal_fcn_new() {
 }
 
 R_API RList *r_anal_fcn_list_new() {
-	RList *list = r_list_new ();
-	if (!list) {
-		return NULL;
-	}
-	list->free = &r_anal_fcn_free;
-	return list;
+	return r_list_new ();
```
Comment (radare) [inline libr/anal/fcn.c]: thats not the same. should be `r_list_newf (r_anal_fcn_free);` this patch should be leaking more

File: libr/core/cmd_help.c [hunk 2]
```diff
@@ -967,8 +967,9 @@ static int cmd_help(void *data, const char *input) {
 		break;
 	case 'w': { // "?w"
 		ut64 addr = r_num_math (core->num, input + 1);
-		const char *rstr = core->print->hasrefs (core->print->user, addr, true);
+		char *rstr = core->print->hasrefs (core->print->user, addr, true);
 		r_cons_println (rstr);
```
Comment (radare) [inline libr/core/cmd_help.c]: check for null. this may crash

File: libr/core/cmd_print.c [hunk 3]
```diff
@@ -5345,14 +5345,15 @@ static int cmd_print(void *data, const char *input) {
 						// XXX: this only works in little endian
 						withref = 0;
 						if (core->print->hasrefs) {
-							const char *rstr = core->print->hasrefs (core->print->user, val, true);
+							char *rstr = core->print->hasrefs (core->print->user, val, true);
 							if (rstr && *rstr) {
 								char *ns; // r_str_ansi_trim (ns, -1, 0);
```
Comment (radare) [inline libr/core/cmd_print.c]: define and assign in the same line

File: libr/core/core.c [hunk 4]
```diff
@@ -2201,8 +2201,8 @@ R_API bool r_core_init(RCore *core) {
 	core->print->exists_var = exists_var;
 	core->print->disasm = __disasm;
 	core->print->colorfor = (RPrintColorFor)r_core_anal_optype_colorfor;
-	core->print->hasrefs = (RPrintColorFor)r_core_anal_hasrefs;
-	core->print->get_comments = (RPrintCommentCallback) r_core_anal_get_comments;
+	core->print->hasrefs = (RPrintHasRefs)r_core_anal_hasrefs;
```
Comment (radare) [inline libr/core/core.c]: 😱

File: libr/egg/egg.c [hunk 5]
```diff
@@ -519,8 +519,8 @@ R_API void r_egg_finalize(REgg *egg) {
 			r_egg_append_bytes (egg, b->buf, b->length);
 		} else {
 			// TODO: use r_buf_cpy_buf or what
-			if (b->length+b->cur > egg->bin->length) {
-				eprintf ("Fuck this shit. Cannot patch outside\n");
+			if (b->length + b->cur > egg->bin->length) {
+				eprintf ("Cannot patch outside\n");
```
Comment (radare) [inline libr/egg/egg.c]: politeness matters

File: libr/core/core.c (hunk 4 above)
Comment (ret2libc) [reply libr/core/core.c]: what?

File: libr/anal/fcn.c (hunk 1 above)
Comment (ret2libc) [reply libr/anal/fcn.c]: please see this commit message https://github.com/radare/radare2/pull/11674/commits/33815683e612bacb0e7aee17ebf0aab105ab7f83

File: libr/core/core.c (hunk 4 above)
Comment (radare) [reply libr/core/core.c]: scary cast, copypasta error

Comment (ret2libc) [reply libr/core/core.c]: I see all functions are assigned this way... shall I remove the cast for all these function pointers?

**Comment by radare** [comment]
well without it we will catch a warning in case the signature doesnt matches

File: libr/anal/fcn.c (hunk 1 above)
Comment (radare) [reply libr/anal/fcn.c]: ?

Comment (ret2libc) [reply libr/anal/fcn.c]: Ok I got it. Fixing it.

**Comment by ret2libc** [comment]
Thanks @radare see it now

**Comment by ret2libc** [comment]
Ok @radare I think this is ready.

**Pull request merged**